### ISSUE TITLE
Feature request: Correct userdata directory on MacOS

### ISSUE THREAD

**Issue by el-dee** [panda3d/panda3d#649]
On MacOS, the method get_user_appdata_directory() of Filename and configuration variable $USER_APPDATA return '~/Documents' which is not right according to the guidelines from Apple (and actually a bit weird to have application data and log suddenly appears in your Document directory...)

The proper location for data is ~/Library/\<application-name>/

To keep backward compatibility with older apps, I would suggest to add a new parameter in prc to select the behavior of get_user_appdata_directory()

On a side note, on Mac and Linux the application data and application config should not be stored in the same directory : On Mac configuration should be under ~/Library/Preferences/\<application-name>/ and on Linux under XDG_CONFIG_DIR, i.e. ~/.config/\<application-name>
 and not under XDG_DATA_DIR, ~/.local/share

But that would require adding a new method in Filename, like get_user_appconfig_dir()

**Comment by Moguri** [comment]
I wonder if this should be combined with #270?

**Comment by paustian** [comment]
I was looking to do this one, but it appears it's already done. Should this be closed?

**Comment by el-dee** [comment]
Actually it is fixed for Panda3D core but not for the dist module, it still maps USER_APPDATA to the ~/Documents directory explicitly.

Also, on macOS the logs should not be stored in ~/Library/<application-name>/ but rather in ~/Library/Logs/<application-name>/ However, there is no concept of USER_APPLOG or something like that in Panda3D so it would require much more work to support it.

**Comment by rdb** [comment]
I agree we ought to have separate directories exposed for config vs cache.

I'm a little nervous that changing these paths is a significant breaking change, since games will no longer be able to find savegames.  My thinking is that perhaps we should add new methods with the correct paths while leaving `get_user_appdata_directory()` alone (perhaps deprecate it).

If someone could work out a proposal for which sets of directories we should offer (in a way that will be most cross-platform), that would be great.
